crates/tauri/scripts/ipc-protocol.js

```diff
@@ -27,16 +27,25 @@
       && (canUseCustomProtocol || cmd === fetchChannelDataCommand)
     ) {
       const { contentType, data } = processIpcMessage(payload)
+      const headers = new Headers({
+        'Content-Type': contentType,
+        'Tauri-Callback': callback,
+        'Tauri-Error': error,
+        'Tauri-Invoke-Key': __TAURI_INVOKE_KEY__
+      })
+      if (options?.headers) {
+        const overrideHeaders =
+          options.headers instanceof Headers
+            ? options.headers.entries()
+            : Object.entries(options.headers)
+        for (const [key, value] of overrideHeaders) {
+          headers.set(key, value)
+        }
+      }
```

Comment:
```suggestion
      const headers = new Headers(options?.headers);
      headers.set("Content-Type", contentType);
      headers.set("Tauri-Callback", callback);
      headers.set("Tauri-Error", error);
      headers.set("Tauri-Invoke-Key", __TAURI_INVOKE_KEY__);
```

I'm just a bit curious, why not implement it this way? It seems more elegant.

I guess it's to preserve previous behavior (allowing users to override preset headers)? But overriding these headers seems like an unreasonable behavior.

Reply:
> I guess it's to preserve previous behavior (allowing users to override preset headers)?

Yeah, that's the reason why I did it like this

> But overriding these headers seems like an unreasonable behavior.

That's true though

I'm not entirely sure if it's supposed to be able to override these or not, cc @lucasfernog

Reply:
welllll nice catch.. that's used internally so we shouldn't allow them to be overwritten (it would break the IPC if you did so - never running or responding depending on which value you change)

Reply:
@Legend-Master can you take his suggestion and change `new Headers(options?.headers)` to `new Headers((options && options.headers) || {})` (null cannot be used in that constructor)

Reply:
pushed :)